pkg/model/crd.go

```diff
@@ -387,6 +387,34 @@ func (r *CRD) GetOutputWrapperFieldPath(
 	return &opConfig.OutputWrapperFieldPath
 }
 
+// OutputShapeName returns name of the Output shape for given operation.
+func (r *CRD) UnwrappedOutputShapeName(
+	// The operation to look for the Output shape
+	op *awssdkmodel.Operation,
+) *string {
+	if op == nil {
+		return nil
+	}
+
+	outputShape := op.OutputRef.Shape
+	if outputShape == nil {
+		return nil
+	}
+
+	// We might be in a "wrapper" shape. Unwrap it to find the real object
+	// representation for the CRD's op. If there is a single member
+	// shape and that member shape is a structure, unwrap it.
+	if outputShape.UsedAsOutput && len(outputShape.MemberRefs) == 1 {
+		for _, memberRef := range outputShape.MemberRefs {
+			if memberRef.Shape.Type == "structure" {
+				outputShape = memberRef.Shape
+				break
+			}
+		}
+	}
+	return &outputShape.ShapeName
```

Comment:
what if `memberRef.Shape.Type` is not "structure" ? L415 will panic
Maybe we can return an error when `outputShape` is not found ?

Reply:
> what if `memberRef.Shape.Type` is not "structure" ? L415 will panic
> Maybe we can return an error when `outputShape` is not found ?

Agreed. This should be `return nil` and the above line 410 should be `return memberRef.Shape`

Reply:
Thanks for the comments, I have updated the code.
I referred latest set_resource.go logic and moved reusable method into CRD.